### ISSUE TITLE
[Suggestion] Allow success roll to trigger alternate Cost on success/fail

### ISSUE THREAD

**Issue by kbrownridge** [crnormand/gurps#473]
Currently we can do this:
[S:Traps ? "You notice something.", "Everything is fine."]
This is awesome.
And now we can do this:
[S:Traps ? "You notice something.", "Everything is fine." *Cost 1 FP]
Also awesome.

What would be even more awesome is this:
[S:Traps ? "You notice something." *Cost 2 FP, "Everything is fine." *Cost 1 FP]

This would be for spells or powers that cost a minimum FP per attempt and full FP on success.

**Comment by crnormand** [comment]
This is starting to go into language development... and I don't know if I want to go down that road at the moment.   One of the issues is that a bunch of our earlier code is not thread safe, mainly anything to do with the modifier bucket.   So items can execute out of order.   I need to fix that before I can continue to make the chat macro "language" more powerful.

But I will keep this open.   It is an interesting idea.

**Comment by kbrownridge** [comment]
Yeah, sorry about that. I'm a web developer for the provincial government by day and GURPS GM by night, so they sometimes spill over.
I do have to say that you are doing an amazing job with this module and it's a joy to use it.
I look forward to seeing where it goes next!

**Comment by cefasheli** [comment]
Open a "trigger" window will be nice  to users who want to use visual modules. They only can triggered by now when we use macros.

**Comment by cefasheli** [comment]
How to work this with Gurps.execute.OTF? 

i need something like 

gurps.execute.OTF( Skill ? "macro" , "macro "),

**Comment by crnormand** [comment]
If `Skill` is a (boolean) variable, then the Javascript code would look like:

`GURPS.executeOTF( Skill ? "macro1" : "macro2")`

If you are trying to determine if the current character has that skill, and if so, execute macro1, otherwise execute macro2:

`GURPS.executeOTF( GURPS.findSkillSpell( GURPS.LastActor, "Skillname") ? "macro1" : "macro2")`

If you are trying to see if the current actor successfully makes the skill roll, and then depending on the skill roll, execute one of the macros... we can't do that.

**Comment by cefasheli** [comment]
Could you explain what is a boolean? i'm just a curious not a programmer (yet) hahahah

**Comment by mjeffw** [comment]
> Could you explain what is a boolean? i'm just a curious not a programmer (yet) hahahah

"Boolean" is a value that must be either "true" or "false".

If you see the following in Javascript, it means "if (a is true) do X; otherwise do Y".

`a ? X : Y`

Chris suggested the following: `GURPS.executeOTF( GURPS.findSkillSpell( GURPS.LastActor, "Skillname") ? "macro1" : "macro2"))`.

To understand this, you have to think like this:

`GURPS.executeOTF(macro)` -- I want to execute a macro. Which macro? That depends on what's inside the parentheses.

Inside the parentheses he put: `GURPS.findSkillSpell( GURPS.LastActor, "Skillname") ? "macro1" : "macro2")`. -- This means, "for the selected character (`GURPS.LastActor`), see if it has a skill named 'Skillname'. If it does, use 'macro1', otherwise use 'macro2'.
